### PR TITLE
[MKT-458]: feat/preBlackFridayLP

### PR DESCRIPTION
**Pull request by jaaaaavier**
Updating Black Friday Landing Page for the content before the Black Friday

### CODE REVIEW

File: src/components/layout/components/LanguageBox.tsx [hunk 1]
```diff
@@ -1,148 +1,156 @@
...
+        size={20}
+        className={`${darkMode ? 'text-white' : 'text-gray-40'} translate-y-px transition duration-150 ease-in-out`}
+      />
+
+      {/* Menu items */}
+      <div className="pointer-events-none absolute top-full left-1/2 z-50 w-52 -translate-x-1/2 translate-y-0 rounded-xl border border-black border-opacity-5 bg-white p-1.5 opacity-0 shadow-subtle transition duration-150 ease-in-out group-hover:pointer-events-auto group-hover:translate-y-1 group-hover:opacity-100">
+        <div className="absolute -top-4 left-1/2 h-4 w-4/5 -translate-x-1/2" />
+
+        <div className="relative grid gap-0 whitespace-nowrap lg:grid-cols-1">
+          {isBlackFriday ? (
```
Comment (jaaaaavier) [inline src/components/layout/components/LanguageBox.tsx]: Changing french for spanish and russian

File: src/components/black-friday/BestStorageSection.tsx [hunk 2]
```diff
@@ -1,38 +1,35 @@
...
+        <div className="top-24 flex flex-col items-center space-y-5">
+          <h2 className="max-w-[526px] text-4xl font-semibold text-white md:text-5xl">
+            {textContent.BestStorage.title}
+          </h2>
+          <h3 className="text-xl text-white">{textContent.BestStorage.subtitle}</h3>
+        </div>
+        <div className="py-9">
+          <ButtonDeal lang={lang} />
+        </div>
+        <Image
```
Comment (jaaaaavier) [inline src/components/black-friday/BestStorageSection.tsx:19]: Changing image resource

File: src/components/black-friday/components/ButtonDeal.tsx [hunk 3]
```diff
@@ -1,18 +1,19 @@
...
+const ButtonDeal = ({ lang }) => {
+  return (
+    <>
+      <button
+        className="relative flex flex-row items-center justify-center rounded-lg bg-primary py-3 px-6 text-base text-white transition duration-100 focus:outline-none focus-visible:bg-primary-dark active:bg-primary-dark sm:text-lg"
+        onClick={() => {
+          window.location.href = '/pricing';
+        }}
+      >
+        {lang === 'es' ? "Elige un plan" : lang === 'ru' ? "Выбрать план" : 'Choose a plan'}
```
Comment (jaaaaavier) [inline src/components/black-friday/components/ButtonDeal.tsx]: New text, for spanish an russian

File: src/components/black-friday/HeroSection.tsx [hunk 4]
```diff
@@ -4,29 +4,36 @@ import ButtonDeal from '@/components/black-friday/components/ButtonDeal';
...
 import { getImage } from '@/lib/getImage';
+import Image from 'next/image';
 
 const HeroSection = ({ textContent, lang }) => {
   const features = [
     {
-      id: 0,
-      text: 'Encrypted file storage and sharing',
-      frenchText: 'Stockage et partage de fichiers cryptés',
+      "id": 0,
```
Comment (xabg2) [inline src/components/black-friday/HeroSection.tsx]: Will this be only in English, Spanish and Russian? Why?

File: src/components/black-friday/HeroSection.tsx [hunk 5]
```diff
@@ -36,42 +43,41 @@ const HeroSection = ({ textContent, lang }) => {
...
           </div>
           <div className="flex w-full  flex-col md:mt-0 md:max-w-none md:flex-row">
             <div className="relative flex items-center justify-center md:h-[600px] lg:left-16 lg:mt-10">
-              <div className="hidden lg:mt-24 lg:flex">
-                <img src={getImage('/images/special-offer/black-friday/file_icons.png')} alt="File icons" />
-              </div>
-              <div className="m-auto flex max-h-[300px] max-w-[300px] md:absolute">
-                <img src="/images/special-offer/black-friday/inxtBF.png" alt="inxtBF image" />
-              </div>
+              <Image src={getImage('/images/black-friday/internxt_black_friday_2024.webp')} alt={'BFCampaign'} width={520} height={520} />
```
Comment (xabg2) [inline src/components/black-friday/HeroSection.tsx:32]: Use better alt for this image.

File: src/components/black-friday/PlatformSection.tsx
Comment (xabg2) [inline src/components/black-friday/PlatformSection.tsx:1]: You may be able to use the PlatformSection component that was created for another page, so you can delete this one and avoid duplication.

File: src/components/black-friday/SuiteSection.tsx [hunk 6]
```diff
@@ -1,70 +1,65 @@
...
+              <div
+                key={index}
+                className="flex flex-col items-start justify-between overflow-hidden rounded-2xl bg-cool-gray-100 lg:grid lg:grid-cols-2 lg:grid-rows-1 lg:gap-0 lg:h-[480px]"
+              >
+                <div className="w-auto px-10 pt-10 lg:h-full lg:w-[480px] lg:p-20 flex flex-col justify-between">
+                  <div className="flex flex-col space-y-4">
+                    <h4 className="text-4xl font-medium lg:text-4xl pb-4">{section.title}</h4>
+                    <h5 className="text-xl">{section.subtitle}</h5>
+                  </div>
+                  <button
```
Comment (xabg2) [inline src/components/black-friday/SuiteSection.tsx]: Use `<Link>` instead (`next/link`)

File: src/components/black-friday/SuiteSection.tsx [hunk 7]
```diff
@@ -1,70 +1,65 @@
...
+                  >
+                    {section.cta}
+                  </button>
+                </div>
+
+                <div className="lg:pl-15 relative mt-16 flex self-stretch lg:mt-0">
+                  <div className="hidden lg:flex lg:max-w-[480px]">
+                    <Image
+                      src={getImage(images[section.image])}
+                      alt='Image'
```
Comment (xabg2) [inline src/components/black-friday/SuiteSection.tsx]: Update alt for this image.

File: src/components/black-friday/TestimonialsSection.tsx
Comment (xabg2) [inline src/components/black-friday/TestimonialsSection.tsx:1]: Use the TestimonialsSection component we have for the main page. Do not duplicate it. Perhaps you can move the component to the shared folder and use it in all the LPs that need it.

File: src/components/black-friday/components/ButtonDeal.tsx (hunk 3 above)
Comment (xabg2) [inline src/components/black-friday/components/ButtonDeal.tsx]: Only 3 langs?

File: src/pages/black-friday.tsx [hunk 8]
```diff
@@ -5,58 +5,98 @@ import HeroSection from '@/components/black-friday/HeroSection';
...
-      <Navbar
-        lang={lang}
-        darkMode={true}
-        isBlackfriday={true}
-        textContent={navbarLang}
-        cta={['Hide Login']}
-        isLinksHidden
-      />
-      <LoginBFBanner />
+      <TopBanner />
```
Comment (xabg2) [inline src/pages/black-friday.tsx]: Why? You can remove this one and use the banner we have in `Layout.tsx`. Then modifying the shouldShowBanner constant like:
`const shouldShowBanner = pathname === '/black-friday' && dialogIsOpen(GlobalDialog.TopBanner);`

You should be able to show the banner only for the black-friday LP.

**Review by xabg2 — CHANGES_REQUESTED** [review]

File: src/components/black-friday/HeroSection.tsx [hunk 9]
```diff
@@ -4,74 +4,111 @@ import ButtonDeal from '@/components/black-friday/components/ButtonDeal';
 import Countdown from '@/components/components/Countdown';
 import Header from '@/components/shared/Header';
 import { getImage } from '@/lib/getImage';
+import Image from 'next/image';
 
 const HeroSection = ({ textContent, lang }) => {
+  const langMapping = {
```
Comment (xabg2) [inline src/components/black-friday/HeroSection.tsx]: Why? You can use the custom hook provided by next called useRouter to get the language.

File: src/components/black-friday/HeroSection.tsx [hunk 10]
```diff
@@ -4,74 +4,111 @@ import ButtonDeal from '@/components/black-friday/components/ButtonDeal';
...
+    en: 'text',
+    es: 'spanishText',
+    ru: 'russianText',
+    it: 'italianText',
+    de: 'germanText',
+    fr: 'frenchText',
+    zh: 'zhText',
+    'zh-tw': 'zhTWText'
+  };
   const features = [
```
Comment (xabg2) [inline src/components/black-friday/HeroSection.tsx]: Move this to the JSONs

File: src/components/black-friday/SuiteSection.tsx [hunk 11]
```diff
@@ -1,70 +1,67 @@
...
+    drive: '/images/privacy/Internxt-Drive.webp',
+    business: '/images/black-friday/internxt_for_families.webp',
+    family: '/images/black-friday/internxt_for_business.webp',
+  };
+  return (
+     <section className="relative flex w-full flex-col bg-highlight">
+      <div className="z-10 flex flex-col items-center py-16">
+        <div className="px-6 text-left sm:text-center">
+          <div className="flex flex-col items-center justify-center px-5 pb-16 text-center">
+            <h2 className="max-w-[756px] text-4xl font-semibold text-white md:text-5xl -mt-2"> {/* Ajustar el margen superior aquí */}
```
Comment (xabg2) [inline src/components/black-friday/SuiteSection.tsx]: Remove the comment

File: src/components/black-friday/components/ButtonDeal.tsx [hunk 12]
```diff
@@ -1,18 +1,30 @@
...
+    fr: "Choisissez un plan",
+    de: "Wählen Sie einen Plan",
+    it: "Scegli un piano",
+    zh: "选择一个计划",
+    "zh-tw": "選擇一個計劃"
+  };
+  const selectedText = translations[lang] || translations.en;
+  return (
+    <>
+      <button
```
Comment (xabg2) [inline src/components/black-friday/components/ButtonDeal.tsx]: Use Link here from next/link and remove the unnecessary <></>

File: src/pages/black-friday.tsx [hunk 13]
```diff
@@ -38,25 +60,47 @@ const BlackFriday = ({ lang, metatagsDescriptions, langJson, navbarLang }) => {
...
+      <MinimalFooter footerLang={footerLang.FooterSection} lang={lang} bgColor='bg-highlight' textColor='text-gray-50' logoColor='white' />
     </Layout>
   );
 };
 
 export async function getServerSideProps(ctx) {
   const language = ctx.locale;
 
-  const allowedLanguages = ['en', 'fr'];
+  const allowedLanguages = ['en', 'es', 'ru', 'zh', 'zh-tw', 'it', 'de', 'fr'];
```
Comment (xabg2) [inline src/pages/black-friday.tsx]: Remove this. If you allow all languages, this array is redundant.

**Review by xabg2 — CHANGES_REQUESTED** [review]

File: src/pages/black-friday.tsx [hunk 14]
```diff
@@ -38,27 +60,46 @@ const BlackFriday = ({ lang, metatagsDescriptions, langJson, navbarLang }) => {
...
   );
 };
 
 export async function getServerSideProps(ctx) {
   const language = ctx.locale;
-
-  const allowedLanguages = ['en', 'fr'];
-
-  const lang = allowedLanguages.includes(language) ? language : 'en';
+  const lang =ctx.locale;
```
Comment (xabg2) [inline src/pages/black-friday.tsx:101]: Use language or lang, but not both.

**Review by xabg2 — CHANGES_REQUESTED** [review]

**Review by xabg2 — APPROVED** [review]

**Pull request merged**